aten/src/ATen/native/Convolution.cpp

```diff
@@ -1485,4 +1490,253 @@ std::tuple<Tensor,Tensor,Tensor> _convolution_double_backward( const c10::option
+      Tensor columns = compute_columns3d(input, params, kernel_size, params.groups,
+                                         /*n_output_plane=*/ weight.size(1));
+      Tensor fgrad_input = at::empty({0}, input.options());
+      std::tie(backend_grad_input, backend_grad_weight, backend_grad_bias) =
+        at::slow_conv3d_backward(grad_output, input, weight, kernel_size, params.stride, params.padding,
+          columns, fgrad_input, output_mask);
+      break;
+    }
+    // Handle backends that don't natively support groups > 1.
+    case ConvBackend::NnpackSpatial:
+    case ConvBackend::Slow2d:
+    case ConvBackend::SlowDilated2d:
+    case ConvBackend::SlowDilated3d:
+    case ConvBackend::SlowTranspose2d:
+    case ConvBackend::SlowTranspose3d:
+    {
+      if (params.groups == 1) {
+        std::tie(backend_grad_input, backend_grad_weight, backend_grad_bias) =
+          _convolution_backward_nogroup_backend(
+            grad_output, input, weight, bias, output_mask, backend, params);
+      } else {
+        std::vector<Tensor> backend_grad_inputs(params.groups);
+        std::vector<Tensor> backend_grad_weights(params.groups);
+        std::vector<Tensor> backend_grad_biases(params.groups);
+        for (int g = 0; g < params.groups; ++g) {
+          auto grad_output_g = subtensor(grad_output, 1, params.groups, g);
+          auto input_g = subtensor(input, 1, params.groups, g);
+          auto weight_g = subtensor(weight, 0, params.groups, g);
+          auto bias_g = subtensor(bias, 0, params.groups, g);
+          std::tie(backend_grad_inputs[g], backend_grad_weights[g], backend_grad_biases[g]) =
+            _convolution_backward_nogroup_backend(
+              grad_output_g, input_g, weight_g, bias_g, output_mask, backend, params);
+        }
+        if (output_mask[0]) {
+          backend_grad_input = at::cat(backend_grad_inputs, 1);
+        }
+        if (output_mask[1]) {
+          backend_grad_weight = at::cat(backend_grad_weights, 0);
+        }
+        if (output_mask[2]) {
+          backend_grad_bias = at::cat(backend_grad_biases, 0);
+        }
+      }
+      break;
+    }
+    // Backward is not supported for these backends.
+    case ConvBackend::Winograd3x3Depthwise:
+      TORCH_CHECK(false, "Backward is not supported for depthwise 3x3 winograd");
+      break;
+    case ConvBackend::Xnnpack2d:
+      TORCH_CHECK(false, "Backward is not supported for xnnpack");
+      break;
+  }
+
+  if (output_mask[0]) {
+    if (k == 3) {
+      backend_grad_input = view3d(backend_grad_input);
+    }
+  }
+  if (output_mask[1]) {
+    if (k == 3) {
+      backend_grad_weight = view3d(backend_grad_weight);
+    }
+  }
+  if (output_mask[2]) {
+    if (bias.defined() && !backend_grad_bias.defined()) {
+      // Calculate bias gradients outside of the backend for those that don't support it.
+      backend_grad_bias = at::empty_like(bias);
+      auto sum_dims = (dim == 3) ? IntArrayRef{0, 2, 3, 4} : IntArrayRef{0, 2, 3};
```

Comment:
I'm not sure this is OK from a UB perspective: IntArrayRef is not owning so it doesn't own the data